### PR TITLE
feat(ST-107): implement endpoint for fetching students for specific classroom

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Added endpoint for fetching student for specific classroom in enrollments.controller file
- Created enrollments.serializer.ts file for trimming the response
- Created interfaces and stored it in enrollments.interface.ts
- Added the logic in the enrollments.service file

## Associated Ticket Link(s)

- https://trello.com/c/uW4bC6Rn

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: src/enrollments/enrollments.serializer.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      "student.user.password",
+      "student.user.createdAt",
+      "student.user.updatedAt",
+      "student.user.student",
+      "student.user.userType",
+      "student.user.id",
+      "classroom",
+    ],
+    populate: ["student", "student.user"],
+  };
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.serializer.ts]: serializeMany should not be so crowded and it should allow most of the attributes. I'd recommend that you use the serialize method that mikro provides inside the controller if this level of custom serialization is necessary.

File: src/enrollments/enrollments.controller.ts [hunk 2]
```diff
@@ -28,4 +33,11 @@ export class EnrollmentsController {
   ): Promise<void> {
     return this.enrollmentsService.removeStudent(user.id, deleteEnrollDto);
   }
+
+  @Roles(EUserType.TEACHER)
+  @Get("students/:classroomId")
+  async getStudentsForClassroom(@Param("classroomId") classroomId: number): Promise<IEnrollment[]> {
+    const enrollments = await this.enrollmentsService.getStudentsForClassroom(classroomId);
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.controller.ts:40]: Let's add new line below

File: src/enrollments/enrollments.controller.ts [hunk 3]
```diff
@@ -28,4 +33,11 @@ export class EnrollmentsController {
...
+    return this.enrollmentSerializer.serializeMany(enrollments);
+  }
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.controller.ts]: I think this should be inside the classrooms controller

Comment (ghost) [reply src/enrollments/enrollments.controller.ts]: This is fetching "enrolled" students for specific classroom, that's why I have decided to include it in this controller.

File: src/enrollments/enrollments.controller.ts [hunk 4]
```diff
@@ -28,4 +33,12 @@ export class EnrollmentsController {
   ): Promise<void> {
     return this.enrollmentsService.removeStudent(user.id, deleteEnrollDto);
   }
+
+  @Roles(EUserType.TEACHER)
+  @Get("students/:classroomId")
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.controller.ts:38]: Please follow REST principles in the tech debt ticket

File: src/enrollments/enrollments.service.ts [hunk 5]
```diff
@@ -65,4 +66,16 @@ export class EnrollmentsService {
...
+  async getStudentsForClassroom(classroomId: number): Promise<IEnrollment[]> {
+    const classroom = await this.classroomRepository.findOneOrFail({ id: classroomId });
+
+    const enrollments = await this.enrollmentRepository.find(
+      { classroom },
+      {
+        populate: ["student", "student.user"],
+      },
+    );
+    return enrollments;
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.service.ts:79]: You might not need the extra variable. Instead, just use return and it should resolve the promise

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left some comments

File: src/enrollments/enrollments.service.ts (hunk 5 above)
Comment (ghost) [reply src/enrollments/enrollments.service.ts:79]: Bhaiya, I do that because there is a lint warning for return and await in the same line.

Comment (ishmam-iqbal-sazim) [reply src/enrollments/enrollments.service.ts:79]: Yes and that's because await with return is redundant as return resolves the promise

**Pull request merged**